packages/manager/src/features/Firewalls/FirewallDetail/Devices/AddNodebalancerDrawer.tsx

```diff
@@ -0,0 +1,181 @@
+      selectedDeviceIds.map((id) => addDevice({ id, type: 'nodebalancer' }))
+    );
+
+    results.forEach((result, _) => {
+      if (result.status === 'fulfilled') {
+        // Assuming the response contains the device label, replace with the appropriate property if not.
+        const label = result.value.entity.label;
+        enqueueSnackbar(`${label} added successfully.`, { variant: 'success' });
+      } else {
+        // Assuming the error object contains the device label, replace with the appropriate property if not.
+        const errorLabel = result.reason.label;
+        enqueueSnackbar(`Failed to add ${errorLabel}.`, { variant: 'error' });
+      }
+    });
+
+    onClose();
+    setSelectedDeviceIds([]);
+  };
+
+  const errorMessage = error
+    ? getAPIErrorOrDefault(error, `Error adding Nodebalancer`)[0].reason
+    : undefined;
+
+  // @todo update regex once error messaging updates
+  const errorNotice = (errorMsg: string) => {
+    // match something like: Nodebalancer <nodebalancer_label> (ID <nodebalancer_id>)
+    const device = /(nodebalancer) (.+?) \(id ([^()]*)\)/i.exec(errorMsg);
+    const openTicket = errorMsg.match(/open a support ticket\./i);
+    if (openTicket) {
+      errorMsg = errorMsg.replace(/open a support ticket\./i, '');
+    }
+    if (device) {
+      const [, label, id] = device;
+      const labelIndex = errorMsg.indexOf(label);
+      errorMsg = errorMsg.replace(/\(id ([^()]*)\)/i, '');
+      return (
+        <Notice
+          sx={{
+            fontFamily: theme.font.bold,
+            fontSize: '1rem',
+            lineHeight: '20px',
+          }}
+          variant="error"
+        >
+          {errorMsg.substring(0, labelIndex)}
+          <Link to={`/nodebalancers/${id}`}>{label}</Link>
+          {errorMsg.substring(labelIndex + label.length)}
+          {openTicket ? (
+            <>
+              <SupportLink text="open a Support ticket" />.
+            </>
+          ) : null}
+        </Notice>
+      );
+    } else {
+      return <Notice text={errorMessage} variant="error" />;
+    }
+  };
+
+  // If a user is restricted, they can not add a read-only Nodebalancer to a firewall.
+  const readOnlyNodebalancerIds = isRestrictedUser
+    ? getEntityIdsByPermission(grants, 'nodebalancer', 'read_only')
+    : [];
+
+  return (
+    <Drawer
+      onClose={() => {
+        setSelectedDeviceIds([]);
+        onClose();
+      }}
+      open={open}
+      title={`Add Nodebalancer to Firewall: ${firewall?.label}`}
+    >
+      <Notice variant={'warning'}>
+        Only the Firewall's inbound rules apply to NodeBalancers. Any existing
+        outbound rules won't be applied.
+        {/* add documentation link */}
+        <Link to="#"> Learn more.</Link>
+      </Notice>
+      <form
+        onSubmit={(e: React.ChangeEvent<HTMLFormElement>) => {
+          e.preventDefault();
+          handleSubmit();
+        }}
+      >
+        {errorMessage ? errorNotice(errorMessage) : null}
```

Comment:
I feel like this could be cleaned up. `errorNotice` shouldn't really need paramaters. 

```suggestion
        {errorMessage ? errorNotice(errorMessage) : null}
```